python/tvm/relay/ty.py

```diff
@@ -107,6 +107,7 @@ class Kind(IntEnum):
     Shape = 3
     Constraint = 4
     AdtHandle = 5
+    # TODO: This is never used.  Remove it?
```

Comment:
I think we should keep it because if we need to implement more complicated kind-checking later, we could ensure nobody is sticking `TypeData` where it shouldn't be (anywhere but the module)